### DISCUSSION TITLE
Is it possible to add full URL of the source page in the saved file's metadata?

### DISCUSSION THREAD

**Discussion by minor6th** [gildas-lormeau/SingleFile#685] [Q&A]
I use SinglePage to clip web pages into DEVONthink on macOS, and I noticed that the saved file has a `com.apple.metadata:kMDItemWhereFroms` metadata item which can be used by DEVONthink.

However there is a problem that it only keeps the source page's domain, it would be great if we can have the full URL of the original web page.
 
I know I could include the info bar in the saved html file, but it cannot be easily processed by productive applications like DEVONthink.

**Comment by gildas-lormeau** [comment]
Extensions cannot change these metadata. Maybe disabling the setting `Misc. > save pages in background` might help though.

**Reply by Uatuu** [reply]
Hi,

SingleFile is very impressive to save web page and keep it as identical as possible from original webpage on a single file.
But even if I know perfection is out of this world, is there any way to add an option to include link to original URL?

My point is people enjoying SingleFile as it is right now will not be impacted, but I'm quite sure there are people like me who would like to have a direct access to original URL from the saved page. I saw this feature on a firefox similar extension I used several years ago (sorry I forgot its name) and it was very useful. Link was present at the very top of page. As I can imagine people willing to store a webpage would like to make sure there is no confusion between original content and any addition that's the reason why I'm suggesting an overlay, but any valid UX will be just great.

Anyway, thank you for this excellent extension.

**Reply by gildas-lormeau** [reply]
Thank you for the kind words! The link to the original URL is displayed in the infobar at the top right of the saved page when displaying it in a browser. If you're using a Chromium-based browser, make sure `Allow access to file URLs` is enabled here `chrome://extensions/?id=mpiodijhokgodhhofbcjdecpffjipkle` in order to display the infobar.

**Reply by Uatuu** [reply]
Alleluia! 
Worst than everything I checked the "Options" first and didn't understood the "Allow access to file URLs"... and was not curious enough to actually try it.
Thank you.

**Reply by gildas-lormeau** [reply]
You're welcome!

**Reply by TomTommieThomson** [reply]
Can you explain what you mean? I don't understand how to view the original URL of the saved HTML file, unfortunately.

**Reply by Uatuu** [reply]
> Can you explain what you mean? I don't understand how to view the original URL of the saved HTML file, unfortunately.

If you want to store the original url in metadata, you need to have the "save url" option to be ticked in "url content" section.
<img width="350" height="280" alt="20260417-09h05m24s-%pn" src="https://github.com/user-attachments/assets/76582f97-5081-41ad-b422-e48a762c783e" />


If you want to directly open original url, make sure you have this option ticked in the "info bar" section.
<img width="351" height="130" alt="20260417-08h55m12s-%pn" src="https://github.com/user-attachments/assets/17bd89bf-4a04-4a06-bdcf-5ea91334cd4f" />

Then, if you click on it, first, you have the saved date, but if you click again on the square arrow icon the original page will load in a new tab.
<img width="947" height="177" alt="20260417-08h50m41s-%pn" src="https://github.com/user-attachments/assets/6d46308c-d356-4568-b37d-51edf5d15379" />
<img width="947" height="177" alt="20260417-08h50m46s-%pn" src="https://github.com/user-attachments/assets/3056efe7-6442-4215-bbbe-0ccec8fb92d2" />

**Reply by TomTommieThomson** [reply]
thank you very much, this is working for me now!
